safeweb/http_test.go

```diff
@@ -488,3 +488,62 @@ func TestRouting(t *testing.T) {
 		})
 	}
 }
+
+func TestGetMoreSpecificPattern(t *testing.T) {
+	for _, tt := range []struct {
+		desc string
+		a    string
+		b    string
+		want string
+	}{
+		{
+			desc: "identical",
+			a:    "/foo/bar",
+			b:    "/foo/bar",
+			want: "",
+		},
+		{
+			desc: "trailing slash",
+			a:    "/foo",
+			b:    "/foo/", // path.Clean will strip the trailing slash.
+			want: "",
+		},
+		{
+			desc: "same prefix",
+			a:    "/foo/bar/quux",
+			b:    "/foo/bar/",
+			want: "/foo/bar/quux",
+		},
+		{
+			desc: "almost same prefix, but not a path component",
+			a:    "/goat/sheep/cheese",
+			b:    "/goat/sheepcheese/",
+			want: "/goat/sheep/cheese",
+		},
+		{
+			desc: "attempt to make less-specific pattern look more specific",
+			a:    "/goat/cat/buddy",
+			b:    "/goat/../../../../../../../cat", // path.Clean catches this foolishness
+			want: "/goat/cat/buddy",
+		},
+		{
+			desc: "2 names for / (1)",
+			a:    "/",
+			b:    "/../../../../../../",
+			want: "",
+		},
+		{
+			desc: "2 names for / (2)",
+			a:    "/",
+			b:    "///////",
+			want: "",
+		},
+	} {
+		t.Run(tt.desc, func(t *testing.T) {
+			got := getMoreSpecificPattern(tt.a, tt.b)
+			if got != tt.want {
+				t.Errorf("%s: got %q, want %q", tt.desc, got, tt.want)
```

Comment:
the subtest name already mentions `tt.desc`
```suggestion
				t.Errorf("got %q, want %q", got, tt.want)
```

Reply:
Done.